app-misc/nnn/nnn-4.7.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 flag-o-matic toolchain-funcs xdg-utils
+
+DESCRIPTION="The missing terminal file browser for X"
+HOMEPAGE="https://github.com/jarun/nnn"
+SRC_URI="https://github.com/jarun/nnn/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="+readline pcre 8contexts icons nerdfonts emoji qsort"
+
+DEPEND="sys-libs/ncurses:=
+	pcre? ( dev-libs/libpcre )
+	readline? ( sys-libs/readline:= )
+	elibc_musl? ( sys-libs/fts-standalone )"
+BDEPEND="virtual/pkgconfig"
+RDEPEND="${DEPEND}"
+REQUIRED_USE="?? ( icons nerdfonts emoji )"
+
+src_prepare() {
+	default
+	tc-export CC
+	use elibc_musl && append-flags "-lfts"
+	sed -i -e '/install: all/install:/' Makefile || die "sed failed"
```

Comment:
Can you please add a comment above the sed why it's necessary? 
(also I see pretty much every distro has this sed, why can't this be fixed upstream? ping @N-R-K )

Reply:
To be honest, I have no idea why that line is necessary. It was apparently added by @Zlogene in 950b512dde3a2fef38e36329d2e8e7b70b8e2bb7 (nnn 3.6).
It replaced the command `sed -i -e '/strip/d' Makefile`. I see why this needed replacing: The `Makefile` changed to include 'strip' in a *lot* of lines. However, I don't see how the `all` target would strip anything, neither in 3.6 nor in 4.7.

The line does not seem to make a difference when building the package, so I removed it.

Reply:
From a quick look, `all` is a prerequisite of `install` so that if someone runs `make install` it without compiling first, the makefile will do the compilation as well.

But I'm not sure what problem it's causing for distro packagers, and I cannot figure anything out from a quick look either, so I'm not sure what to fix here. /shrug

Reply:
Update: That `sed` line did not do anything (missing `s`). However, it *is* necessary in combination with nnn's bundled patches. The `Makefile` applies the patches, compiles and then reverts the patches. That makes the source files newer than the compiled binary. The 'install' target therefore compiles nnn *again*. Since the ebuild does not pass the respective flags `src_install`, that second build was "vanilla", i.e. without any customization at all. That explains why the patches seemed to break the other USE flags.

TL;DR: I added the line back, fixed it and wrote a comment. I also added USE flags for nnn's bundled patches.

Reply:
> Edit: See [jarun/nnn#1493](https://github.com/jarun/nnn/issues/1493).

Ah, I see. I forgot about the patching shenanigans - makes sense now. Though it's still a bit weird why the ebuild was doing it even though it didn't allow patches.

Reply:
Remember we allow user patching via */etc/portage/patches*, I imagine this will override them as well.